python_modules/dagster/dagster/_grpc/proxy_server.py

```diff
@@ -156,13 +166,13 @@ def cleanup(self):
         # In case ShutdownServer was not called
```

Comment:
clean up the server one here too

Reply:
@dpeng817 did you see this one? you want something like this with the server heartbeat thread I think: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_grpc/server.py#L462-L463

Reply:
agh sorry for the miss. that totally makes sense.

Reply:
Added a join to `cleanup`. We don't need to add one to `ReloadCode` though right? We only need to clean up the original client heartbeat thread there since we're polling a new underlying grpc server?

Reply:
that's right, since we're not swapping out the server thread we should not need to join it